lib/tasks/fixes.rake

```diff
@@ -113,6 +113,17 @@ namespace :fixes do
     end
   end
 
+  desc "Regenerate JPEG logos :invoice style"
+  task :regenerate_jpeg_invoice_logo => :environment do
+    Profile.where(logo_content_type: "image/jpeg").find_each do |profile|
+      begin
+        profile.logo.reprocess!(:invoice)
```

Comment:
I am wondering if we are affected by this bug: https://github.com/thoughtbot/paperclip/pull/2637

I have tested this successfully though, so not sure if this may cause any issues. But this is to be aware of for SaaS migration.

Reply:
I think we should be fine. Since we are not running an external process like it was done by ImageMagic, I think in case of a trouble, the process will just fail. Also if target style only is truncated, that shouldn't be an issue.

But I will add as a precaution to backup the S3 bucket beforehand. Good catch!

Reply:
I managed to test this by editing `gems/paperclip-6.1.0/lib/paperclip/storage/s3.rb` and making sure an empty local file is being generated when copying the file locally.

The bug is present - paperclip overwrites the target image with an empty file.

But we have nothing to worry because:
1. it only overwrites the styles being reprocessed, this means that original logo will not be affected
2. because the `invoice` style will be missing anyway, overwriting it with a zero byte file will not matter (no data will be lost)

What we have to do would be to check for any broken invoice logos after running the task.